### PR TITLE
Extract utility TypeDeserializerModule

### PR DESCRIPTION
**Pull request by kokosing**
Extract utility TypeDeserializerModule

### CODE REVIEW

**Review by ksobolew — APPROVED** [review]
Love it! ❤️

I recall that you mentioned that there's still one more instance of this, but it's hard to get rid of - can you add a `TODO` or `FIXME` there for the future?

**Review by gaurav8297 — APPROVED** [review]

**Comment by gaurav8297** [comment]
Looks good from my side!

**Review by Praveen2112 — APPROVED** [review]

File: plugin/trino-google-sheets/pom.xml [hunk 1]
```diff
@@ -129,6 +129,12 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.openjdk.jol</groupId>
+            <artifactId>jol-core</artifactId>
+            <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (kokosing) [reply plugin/trino-google-sheets/pom.xml:135]: It started to complain when I was adding other dependencies. Not sure why it was not complaining in a first place just for this module.

**Comment by kokosing** [comment]
> I recall that you mentioned that there's still one more instance of this, but it's hard to get rid of - can you add a TODO or FIXME there for the future?

The question is whether we should actually fix it. The class is here `io.trino.type.TypeDeserializer`. It is using different guice binding and it is part of the engine. Currently engine also depends on `trino-plugin-toolkit` but I am not sure if it is the way to go for this case.

**Pull request merged**